src/Mod/Fem/femresult/resulttools.py

```diff
@@ -411,11 +411,43 @@ def add_principal_stress_std(res_obj):
     res_obj.PrincipalMin = prinstress3
     res_obj.MaxShear = shearstress
     FreeCAD.Console.PrintLog("Added standard principal stresses and max shear values.\n")
+
+    # add critical strain ratio
+    MatMechNon = FreeCAD.ActiveDocument.getObject('MaterialMechanicalNonlinear')
+    if MatMechNon:
+        stress_strain = MatMechNon.YieldPoints
+        if stress_strain:
+            i = -1
+            while stress_strain[i] == "": i -= 1
+            eps_cr_uni = float(stress_strain[i].split(",")[1])
+            alpha = 1.65 * eps_cr_uni
+            beta = 1.5
+            if res_obj.Peeq:
+                res_obj.CriticalStrainRatio = calculate_csr(prinstress1, prinstress2, prinstress3, alpha, beta,
+                                                            res_obj)
+
     return res_obj
 
 
-def get_concrete_nodes(res_obj):
+def calculate_csr(ps1, ps2, ps3, alpha, beta, res_obj):
+    #
+    # HarryvL: calculate critical strain ratio
```

Comment:
What's the protocol for adding the dev name in the code? (I've seen it appended to the end of the line in parenthesis)

Reply:
This came in at the time of concrete development when I cooperated with Berne. I can remove.

Reply:
Yes, please remove your name, but keep the comment since it explains what "csr" stands for.